### PR TITLE
Allow swift sdk to delete a wildcard

### LINKED ISSUE #19: Allow Swift SDK to delete on wildcards
Server Version: Master
Client Version: Master
Improvement: Allow deletes on wildcards

### PR DESCRIPTION
**Pull request by boopt2**
Resolves #19 
Proposed Changes:
*  Allow user to delete on wildcards
DCO1.1 Signed-off-by: Thomas Boop <tgboop@us.ibm.com>

### CODE REVIEW

**Comment by boopt2** [comment]
Hi boopt2,

Thanks for submitting this pull request!

I can confirm that the DCO1.1 sign-off has been included. It is okay to process this pull request.

dco-bot

File: CSyncSDK/Operation.swift [hunk 1]
```diff
@@ -196,8 +196,8 @@ class PubOperation : Operation
 			return nil
 		}
 
-		guard !key.isKeyPattern else {
-			error = err(CSError.invalidKey, msg:"Key may not contain wildcard characters")
+		guard !key.isKeyPattern || delete == true else {
+			error = err(CSError.invalidKey, msg:"Only delete pubs may contain a wildcard")
```
Comment (mkistler) [inline CSyncSDK/Operation.swift]: I think this error message will be confusing to developers.  The developer does a "write", not a "pub", so a message about "delete pubs" won't help them.  I think you could leave the message as it is, but if you really feel it needs more detail, I would suggest "Key for write may not contain wildcard characters".

File: CSyncSDKTests/ListenTests.swift [hunk 2]
```diff
@@ -156,6 +156,54 @@ class ListenTests: XCTestCase {
...
+					if keyOne && keyTwo && keyThree {
+						expectation.fulfill()
+					}
+				}
+			}
+			let writeKey7 = app.key("tests.DeleteWildcard." + uuid + "a.*")
+			writeKey7.delete()
+		}
+		waitForExpectations(timeout: 20.0, handler:nil)
+	}
```
Comment (mkistler) [inline CSyncSDKTests/ListenTests.swift]: This test does not belong in the `ListenTests` file.  It should be in the `WriteTests` file -- all the other delete tests are in that file.

**Review by mkistler — CHANGES_REQUESTED** [review]
Looks good.  Just needs a few small changes.

**Review by mkistler — APPROVED** [review]
Looks good! 👍

**Pull request merged**